atlasdb-config/src/main/java/com/palantir/atlasdb/http/FailoverFeignTarget.java

```diff
@@ -57,6 +57,7 @@
     public FailoverFeignTarget(Collection<String> servers, Class<T> type) {
         this.servers = ImmutableList.copyOf(ImmutableSet.copyOf(servers));
         this.type = type;
+        this.numServersToTryBeforeFailing = Math.min(servers.size(), 14);
```

Comment:
So servers.size() may be too small.  If you only have 3 servers you may
want to cycle through them several times before you give up.

The way I tuned these numbers originally was how long "the" server took to
start in 2009.  So if you have a single server and restarted it, the client
would stay connected with pretty high probability.  Basically try each
server 3 times and loop through them and add some backoff each time you
visit a new server.  This should be like 30 - 40s which by the time the
server comes back up the client doesn't barf.




On Tue, Aug 23, 2016 at 2:39 PM, Tom Petracca <notifications@github.com>
wrote:

> In atlasdb-config/src/main/java/com/palantir/atlasdb/http/
> FailoverFeignTarget.java
> <https://github.com/palantir/atlasdb/pull/813#discussion_r75956566>:
>
> > @@ -57,6 +57,7 @@
> >      public FailoverFeignTarget(Collection<String> servers, Class<T> type) {
> >          this.servers = ImmutableList.copyOf(ImmutableSet.copyOf(servers));
> >          this.type = type;
> > +        this.numServersToTryBeforeFailing = Math.min(servers.size(), 14);
>
> @carrino <https://github.com/carrino> for magic number 14 enlightenment.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/palantir/atlasdb/pull/813/files/74aca031d01412dfe6a7cf20b876e37f50efc907#r75956566>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA7s26H7NE7AVUODxLYGbXghKafPgekIks5qi2iWgaJpZM4Joo12>
> .
>